### DISCUSSION TITLE
How to model polymorphic one-to-many relationship in prisma?

### DISCUSSION THREAD

**Discussion by shash7** [prisma/prisma#11040] [Q&A] [answered]
Howdy folks, I have 2 tables:

- Order

- Lead

I want to be able to attach a lead to the order. However multiple orders can have different leads.
So Order can have a leadId stored but a single lead can have multiple orders

This is best summarized by this laravel one-to-many poly doc page:
https://laravel.com/docs/8.x/eloquent-relationships#polymorphic-relationships

How would I do this in prisma?

**Comment by austincrim** [comment, marked as answer]
Hey @shash7 👋 ,

Prisma does not currently support "polymorphic" relationships. There has been a lot of previous discussion about this topic you can read about in #1644 or #2506.

I might be misunderstanding your use case, but you could use a plain one-to-many relationship in this scenario?

Thanks!

**Reply by shash7** [reply]
No, a polymorphic one-to-many is very specific.

Imagine a user model who has a couple of posts model attached to it.
In this case, these posts exclusively belong to the user and this is a garden variety one-to-many.

Now imagine a post model which is attached to a categorypivot model which in turn, is attached to a category model.
In this case, the categories don't belong exclusively to a particular post. So you'll need a pivot table(categorypivot) to attach the post to a category. This is a polymorphic one-to-many relation.

Seems like Prisma doesn't support one-to-many poly. However this is a fairly common use case and I'm wondering if anybody here can show me how to model it inside the schema.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
